### PR TITLE
Issue Forms - BETA

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites for Code Changes

* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed

* Replace markdown files to new issue forms.

### Additional Information

This is a proposal changes because they are still in public beta for public repositories. Will not work for private repos. Feel free to request some proposal in this PR. If will be LGTM i can make them to review.

References: 

- https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

### CODE REVIEW

**Review by PhilippHeuer — APPROVED** [review]
LGTM :)

I like this a lot, especially with the new placeholders.

**Comment by stachu540** [comment]
@PhilippHeuer could you clarify, if current required changes is fine with that?

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      description: A clear and concise description of the feature/enhancement.
+      placeholder: e.g. I have an issue when [...]
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Alternatives
+      description: A clear and concise description of any alternative solutions or features you've considered.
+    validations:
+      required: true
```
Comment (PhilippHeuer) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: This should be optional, there may not be alternative solutions at all times.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ## Feature Request
+  - type: checkboxes
+    attributes:
+      label: Prerequisites
+      options:
+        - label: If this is a question/suggestion, I have already considered discuss it on [Discord Server](https://discord.gg/FQ5vgW3)
```
Comment (PhilippHeuer) [inline .github/ISSUE_TEMPLATE/feature_request.yml]: Lets change `considered discuss` to `considered to discuss`

**Review by PhilippHeuer — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ## Bug Report
+  - type: checkboxes
+    attributes:
+      label: Prerequisites
+      description: |
+        The following prerequisites must be filled. If you are spot a bug please provide some details in [Environment](#environment)
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        The following prerequisites must be fulfilled. If you are spot a bug please provide some details in [Environment](#environment)
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      options:
+        - label: If this is a question/suggestion, I have already considered discuss it on [Discord Server](https://discord.gg/FQ5vgW3)
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - label: If this is a question/suggestion, I have already considered discussing it on the [Discord Server](https://discord.gg/FQ5vgW3)
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        - label: I have attempted to [look for similar issues](https://github.com/twitch4j/twitch4j/issues) already
+          required: true
+        - label: The bug have a spotted
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: The meaning of this is unclear

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/feature_request.yml]:
```suggestion
        - label: If this is a question/suggestion, I have already considered discussing it on the [Discord Server](https://discord.gg/FQ5vgW3)
```

**Review by iProdigy — CHANGES_REQUESTED** [review]
Thanks for migrating this! Just a few minor nits

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (iProdigy) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: Superceded by https://github.com/twitch4j/.github/pull/2#discussion_r660202524

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 5 above)
Comment (stachu540) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Author must know if bug have been spotted in specific environment and providing console output with the stacktrace. If one of them not meet requirements, the issue is `invalid` and author must provide more explanation of that bug or maintainers close the issue if it is too much complex or issue author has been inactive in `x` days (marking `stale`).

**Any suggestions to that?**

EDIT: proposal: `The bug have been spotted in specific environment.`

**Comment by stachu540** [comment]
Final checks to review. I think...

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        ## Bug Report
+  - type: checkboxes
+    attributes:
+      label: Prerequisites
+      description: |
+        The following prerequisites must be fulfilled. 
+        If you are spot a bug please provide some details in [Environment](#issue_form_environment) field and [console output with the stacktrace](#issue_form_log_output_stacktrace) (if it is avaliable).
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        If you spot a bug, please provide some details in the [Environment](#issue_form_environment) field and [console output with the stacktrace](#issue_form_log_output_stacktrace) (if it is avaliable).
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      options:
+        - label: If this is a question/suggestion, I have already considered discussing it on the [Discord Server](https://discord.gg/FQ5vgW3)
+        - label: I have attempted to [look for similar issues](https://github.com/twitch4j/twitch4j/issues) already
+          required: true
+        - label: The bug have been spotted in specific environment.
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - label: The bug has been spotted in specific environment.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        
+        dependencies {
+          implementation 'com.github.twitch4j:twitch4j:1.3.1'
+        }
+        ```
+  - type: textarea
+    attributes:
+      label: Possible Solution
+      description: A suggestion for a fix, if you have a idea
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
      description: A suggestion for a fix, if you have an idea
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        - **Java Version**: 
+        - **Build tools**: 
+        - **Operating system**: 
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Build Configuration
+      description: |
+        Paste your build configuration if you are using it. Please use code blocks for specific build.
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        Paste your build configuration if you are using it. Please use code blocks for specific build configs.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+          at com.github.twitch4j.examples.TwitchExampleTrace.methodA(TwitchExampleTrace.java:9)
+          at com.github.twitch4j.examples.TwitchExampleTrace.main(TwitchExampleTrace.java:5)
+      render: txt
+  - type: textarea
+    attributes:
+      label: Environment
+      description: |
+        Present your environment specification to your project. Required to resolving your issue.
+        - **Library Version**: The library version affected to your project. *e.g. 1.2.3, 1.3.1, 1.5.0*
+        - **IDE**: Your integrated development enviornment name with or without the version, or URL if it is web-based cloud IDE. *e.g. IntelliJ IDEA 2021.2, Codeanywhere (https://codeanywhere.com/), VSCode*
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - **IDE**: Your integrated development enviornment name with or without the version, or URL if it is a web-based cloud IDE. *e.g. IntelliJ IDEA 2021.2, Codeanywhere (https://codeanywhere.com/), VSCode*
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        - **Java Version**: Your current java version. `java -version` *e.g. 16.0.1, 1.8.0_281, 11.0.10*
+        - **Build tools**: Your build tools with the version if you are using it. *e.g. Gradle 6.7, Maven 3.3.9* - if you not using it just leave it *empty* or `none`
+        - **Operating system**: Your operating system where you are build it *e.g. Mac OS 11 (M1), Mac OSX 10.13, Windows 10, Ubuntu 20.04*
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - **Operating system**: The operating system where you are building and/or executing the code *e.g. Mac OS 11 (M1), Mac OSX 10.13, Windows 10, Ubuntu 20.04*
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+      value: |      
+        - **Library Version**: 
+        - **IDE**: 
+        - **Java Version**: 
+        - **Build tools**: 
+        - **Operating system**:
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - **Operating System**: 
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        - **Library Version**: The library version affected to your project. *e.g. 1.2.3, 1.3.1, 1.5.0*
+        - **IDE**: Your integrated development enviornment name with or without the version, or URL if it is web-based cloud IDE. *e.g. IntelliJ IDEA 2021.2, Codeanywhere (https://codeanywhere.com/), VSCode*
+        - **Java Version**: Your current java version. `java -version` *e.g. 16.0.1, 1.8.0_281, 11.0.10*
+        - **Build tools**: Your build tools with the version if you are using it. *e.g. Gradle 6.7, Maven 3.3.9* - if you not using it just leave it *empty* or `none`
+        - **Operating system**: Your operating system where you are build it *e.g. Mac OS 11 (M1), Mac OSX 10.13, Windows 10, Ubuntu 20.04*
+      value: |      
+        - **Library Version**: 
+        - **IDE**: 
+        - **Java Version**: 
+        - **Build tools**:
```
Comment (iProdigy) [inline .github/ISSUE_TEMPLATE/bug_report.yml]:
```suggestion
        - **Build Tools**: 
```

**Review by iProdigy — CHANGES_REQUESTED** [review]
last nits!

**Review by iProdigy — APPROVED** [review]
Thank you!

**Review by PhilippHeuer — APPROVED** [review]
i made a few small tweaks, but this can be merged now imo

**Pull request merged**